From 7a1792112f3877366699a65e9d41f2c772e0fa72 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rn=20Kottmann?= <joern@apache.org>
Date: Wed, 28 Sep 2011 11:37:49 +0000
Subject: [PATCH] OPENNLP-288 Now tokens are lower cased when old tag
 dictionary format is used and case sensitive is set to false.

git-svn-id: https://svn.apache.org/repos/asf/incubator/opennlp/trunk@1176832 13f79535-47bb-0310-9956-ffa450edef68
---
 .../src/main/java/opennlp/tools/postag/POSDictionary.java    | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/opennlp-tools/src/main/java/opennlp/tools/postag/POSDictionary.java b/opennlp-tools/src/main/java/opennlp/tools/postag/POSDictionary.java
index e5ca312df..91dde216a 100644
--- a/opennlp-tools/src/main/java/opennlp/tools/postag/POSDictionary.java
+++ b/opennlp-tools/src/main/java/opennlp/tools/postag/POSDictionary.java
@@ -118,7 +118,10 @@ public POSDictionary(BufferedReader reader, boolean caseSensitive) throws IOExce
       for (int ti = 0, tl = parts.length - 1; ti < tl; ti++) {
         tags[ti] = parts[ti + 1];
       }
-      dictionary.put(parts[0], tags);
+      if (caseSensitive)
+        dictionary.put(parts[0], tags);
+      else
+        dictionary.put(parts[0].toLowerCase(), tags);
     }
   }
 
